include/crm/pengine/print_ml_status.h

```diff
@@ -0,0 +1,128 @@
+#define mon_show_fence_history (0x0800U)
+
+#define mon_show_headers       (mon_show_times | mon_show_stack | mon_show_dc \
+                               | mon_show_count)
+#define mon_show_default       (mon_show_headers | mon_show_nodes \
+                               | mon_show_resources)
+#define mon_show_all           (mon_show_default | mon_show_attributes \
+                               | mon_show_failcounts | mon_show_operations \
+                               | mon_show_tickets | mon_show_bans \
+                               | mon_show_fence_history)
+
+/*
+ * Definitions indicating how to output
+ */
+
+enum mon_output_format_e {
+    mon_output_none,
+    mon_output_monitor,
+    mon_output_plain,
+    mon_output_console,
+    mon_output_xml,
+    mon_output_html,
+    mon_output_cgi
+} output_format = mon_output_console;
+
+struct print_params_t
+{
+    unsigned int show; // = mon_show_default;
+    gboolean group_by_node; // = FALSE;
+    gboolean inactive_resources; // = FALSE;
+    gboolean print_timing; // = FALSE;
+    gboolean fence_history; // = FALSE
+    gboolean fence_full_history; // = FALSE
+    gboolean print_brief; // = FALSE;
+    gboolean print_pending; // = TRUE;
+    gboolean print_clone_detail; // = FALSE;
+    /* FIXME allow, detect, and correctly interpret glob pattern or regex? */
+    const char *print_neg_location_prefix; // = "";
+};
+
+/* Define exit codes for monitoring-compatible output
+ * For nagios plugins, the possibilities are
+ * OK=0, WARN=1, CRIT=2, and UNKNOWN=3
+ */
+#define MON_STATUS_WARN    CRM_EX_ERROR
+#define MON_STATUS_CRIT    CRM_EX_INVALID_PARAM
+#define MON_STATUS_UNKNOWN CRM_EX_UNIMPLEMENT_FEATURE
+
+/* Convenience macro for prettifying output (e.g. "node" vs "nodes") */
+#define s_if_plural(i) (((i) == 1)? "" : "s")
+
+#if CURSES_ENABLED
+#  define print_dot() if (output_format == mon_output_console) { \
+        printw(".");                            \
+        clrtoeol();                             \
+        refresh();                              \
+    } else {                                    \
+        fprintf(stdout, ".");                   \
+    }
+#else
+#  define print_dot() fprintf(stdout, ".");
+#endif
+
+#if CURSES_ENABLED
+#  define print_as(fmt, args...) if (output_format == mon_output_console) { \
+        printw(fmt, ##args);                            \
+        clrtoeol();                                     \
+        refresh();                                      \
+    } else {                                            \
+        fprintf(stdout, fmt, ##args);                   \
+    }
+#else
+#  define print_as(fmt, args...) fprintf(stdout, fmt, ##args);
+#endif
+
+int get_resource_display_options(void);
+void print_cluster_summary(FILE *stream, pe_working_set_t *data_set);
+int count_resources(pe_working_set_t * data_set, pe_resource_t * rsc);
+void print_resources(FILE *stream, pe_working_set_t *data_set, int print_opts);
```

Comment:
This would be the only one that requires/exposes "print_opts" from here.  Would it make sense to wrapper this function providing parameters ```enum mon_output_format_e output_format```and probably also ```struct print_params_t *print_params```, so that get_resource_display_options() wouldn't need to be exposed?